Sources/Build/BuildPlan.swift

```diff
@@ -1263,6 +1419,251 @@ public final class SwiftTargetBuildDescription {
+                                    .pathString
+                            )
+                        ]
+                    )
+                )
+            }
+
+            // TODO(ncooke3): What happens if a custom module map exists with a
+            // name other than `module.modulemap`?
+            try VFSOverlay(roots: roots)
+                .write(to: allProductHeadersPath, fileSystem: fileSystem)
+
+            // For Product directory
+            try VFSOverlay(roots: [
+                VFSOverlay.Directory(
+                    name: buildArtifactDirectory
+                        .appending(component: "Product").pathString,
+                    contents:
+                        // Public headers
+                        try VFSOverlay.overlayResources(
+                            directoryPath: publicHeadersPath,
+                            fileSystem: fileSystem,
+                            shouldInclude: {
+                                // Filter out a potential custom module map as
+                                // only the generated module map in the build
+                                // directory is used.
+                                !$0.pathString.hasSuffix("module.modulemap")
+                            }
+                        ) + [
+                            VFSOverlay.File(
+                                name: generatedInteropHeaderPath.basename,
+                                externalContents: generatedInteropHeaderPath.pathString
+                            )
+                        ]
+                )
+            ]).write(to: allProductHeadersPathProduct, fileSystem: fileSystem)
+
+            swiftTargetBuildDescription.additionalFlags += [
+                // Builds Objective-C portion of module.
+                "-import-underlying-module",
+                // Add the location of the module's Objective-C module map as
+                // a header search path.
+                "-I",
+                buildArtifactIntermediatesDirectory.pathString
+            ]
+
+            swiftTargetBuildDescription.appendClangFlags(
+                // Pass VFS overlay to the underlying Clang compiler.
+                "-ivfsoverlay", allProductHeadersPath.pathString,
+                // Pass VFS overlay to the underlying Clang compiler.
+                "-ivfsoverlay", unextendedModuleMapOverlayPath.pathString
+            )
+
+            // Overlay the public headers over the build directory and add
+            // the directory to the header search paths. This will also help
+            // pickup generated headers in the build directory like the
+            // generated interop Swift header.
+            clangTargetBuildDescription.additionalFlags += [
+                "-I",
+                buildArtifactIntermediatesDirectory.pathString,
+                // TODO(ncooke3): #123 This is pretty hacky and doesn't scale well.
```

Comment:
123?

Reply:
I had several TODOs that were referencing the same issue. I was keeping track of it by adding a number so I could search all related comments with `TODO(ncooke3): #123`. My latest commits resolved that issue entirely so this is no more.